### DISCUSSION TITLE
Add support for Pyrogram

### DISCUSSION THREAD

**Discussion by PBhadoo** [cloudflare/workerd#1971] [Python Packages]
Pyrogram is a modern, elegant and asynchronous [MTProto API](https://docs.pyrogram.org/topics/mtproto-vs-botapi) framework. It enables you to easily interact with the main Telegram API through a user account (custom client) or a bot identity (bot API alternative) using Python.

There are several Projects that are inside pypi.org that are similar to Pyrogram based on Pyrogram.

Add some for bringing Telegram Devs to Cloudflare Workers

https://pypi.org/search/?q=pyrogram&o=-created

Official looks like this one
https://pypi.org/project/Pyrogram/

**Comment by Guru322** [comment]
Up

**Comment by rojvv** [comment]
This is technically impossible. Telegram clients aren’t supposed to be run on serverless.

**Reply by dom96** [reply]
Depends on the Telegram API. Discord Bots work just fine on Cloudflare Workers.

**Reply by septatrix** [reply]
MTProto however depends to the best of my knowledge on long polling which is infeasible for workers. The normal HTTP webhook API on the other hand should work but that is not what is used by Pyrogram

**Reply by rojvv** [reply]
@dom96 Cloudflare Workers already [supports](https://grammy.dev/hosting/cloudflare-workers) Telegram’s Bot API—which is simply an HTTP API with support for webhooks. _Telegram clients_ don’t use HTTP and depend on a persistent connection TCP instead with a different protocol.

**Reply by PBhadoo** [reply]
If it's added. Is it possible to use 15 min [Cron Trigger](https://developers.cloudflare.com/workers/configuration/cron-triggers/) ?

**Reply by dom96** [reply]
@roj1512 you can hold a persistent TCP connection via Durable Objects (in a similar way that you can for websocket connections).

**Comment by ryanking13** [comment]
pyrogram project is archived and not maintained anymore. So I don't think  we will support it.
